src/System.IO.Compression/tests/ZipArchive/zip_CreateTests.cs

```diff
@@ -149,10 +149,58 @@ public static void CreateUncompressedArchive()
             }
         }
 
+        [Fact]
+        public static void CreateNormal_VerifyDataDescriptor()
+        {
+            using var memoryStream = new MemoryStream();
+            // We need an non-seekable stream so the data descriptor bit is turned on when saving
+            var wrappedStream = new WrappedStream(memoryStream, true, true, false, null);
+
+            // Creation will go through the path that sets the data descriptor bit when the stream is unseekable
+            using (var archive = new ZipArchive(wrappedStream, ZipArchiveMode.Create))
+            {
+                CreateEntry(archive, "A", "xxx");
+                CreateEntry(archive, "B", "yyy");
+            }
+
+            AssertDataDescriptor(memoryStream, true);
+
+            // Update should flip the data descriptor bit to zero on save
+            using (var archive = new ZipArchive(memoryStream, ZipArchiveMode.Update))
+            {
+                ZipArchiveEntry entry = archive.Entries[0];
+                using Stream entryStream = entry.Open();
+                StreamReader reader = new StreamReader(entryStream);
```

Comment:
technically this Reader is a disable object that will close the stream when disposed.  If the GC decided to collect after the last reference to reader it would close your MemoryStream and your other calls can fail with ObjectDisposed.